File: jdroid-gradle-googleplay-publisher-plugin/src/main/groovy/com/jdroid/android/googleplay/publisher/task/ListTracksTask.java
package com.jdroid.android.googleplay.publisher.task;

import com.google.api.services.androidpublisher.model.Track;
import com.google.api.services.androidpublisher.model.TracksListResponse;
import com.jdroid.android.googleplay.publisher.App;
import com.jdroid.android.googleplay.publisher.GooglePlayPublisher;

import org.gradle.api.logging.LogLevel;

public class ListTracksTask extends BaseTask {
	
	public ListTracksTask() {
		setDescription("List all the assigned APKs for each release track");
	}

	@Override
	protected void onExecute(App app) {
		TracksListResponse tracksListResponse = GooglePlayPublisher.getTracks(app);
		getLogger().log(LogLevel.INFO, "-------------------------------");
		for (Track track : tracksListResponse.getTracks()) {
			getLogger().log(LogLevel.INFO, "Track: " + track.getTrack());
			getLogger().log(LogLevel.INFO, "Version codes: " + track.getVersionCodes());
			getLogger().log(LogLevel.INFO, "User fraction: " + track.getUserFraction());
			getLogger().log(LogLevel.INFO, "-------------------------------");
		}
	}

}

File: google-play-api-plugin/src/main/java/com/dipien/google/play/api/task/ListTracksTask.kt
package com.dipien.google.play.api.task

import com.dipien.google.play.api.App
import com.dipien.google.play.api.GooglePlayPublisher
import org.gradle.api.logging.LogLevel

open class ListTracksTask : BaseTask() {

    private val logLevel = LogLevel.LIFECYCLE

    init {
        description = "List all the tracks and its releases"
    }

    override fun onExecute(app: App) {

        GooglePlayPublisher.getTracks(app).forEach { track ->
            if (track.releases != null) {
                logger.log(logLevel, "Track: " + track.track)
                logger.log(logLevel, "-------------------------------")
                track.releases.forEach { trackRelease ->
                    logger.log(logLevel, "  Release name: " + trackRelease.name)
                    logger.log(logLevel, "  Version codes: " + trackRelease.versionCodes)
                    logger.log(logLevel, "  User fraction: " + trackRelease.userFraction)
                    logger.log(logLevel, "  Status: " + trackRelease.status)
                    logger.log(logLevel, "  Release Notes: " + trackRelease.releaseNotes)
                    logger.log(logLevel, "")
                }
                logger.log(logLevel, "")
            }
        }
    }
}
